File: app/src/main/java/com/example/teprinciple/updateappdemo/MainActivity.java
package com.example.teprinciple.updateappdemo;

import android.content.Intent;
import android.os.Bundle;
import android.os.Environment;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

import org.jetbrains.annotations.NotNull;

import listener.UpdateDownloadListener;
import model.DownLoadBy;
import model.UpdateConfig;
import update.UpdateAppUtils;


public class MainActivity extends AppCompatActivity {

    //服务器apk path,这里放了百度云盘的apk 作为测试
    String apkPath = "http://issuecdn.baidupcs.com/issue/netdisk/apk/BaiduNetdisk_9.6.63.apk";
    private int code = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void updateApp(View view) {
        realUpdate(1);
    }

    public void downloadByWeb(View view) {
        realUpdate(2);
    }

    public void forceUpdate(View view) {
        realUpdate(3);
    }

    public void checkByName(View view) {
        realUpdate(4);
    }

    public void kotlin(View view) {
        realUpdate(5);
    }


    private void realUpdate(int code) {
        this.code = code;
        switch (code) {
            case 1:
                updat1();
                break;
            case 2:
                update2();
                break;
            case 3:
                update3();
                break;
            case 4:
                update4();
                break;
            case 5:
                startActivity(new Intent(this, KotlinDemoActivity.class));
                break;

        }
    }


    //基本更新
    private void updat1() {
        UpdateConfig config = new UpdateConfig();
        config.setDownloadBy(DownLoadBy.APP);
        config.setCheckWifi(true);
        config.setForce(false);
        config.setApkSavePath(Environment.getExternalStorageDirectory().getAbsolutePath() + "/teprinciple");

        UpdateAppUtils
                .getInstance()
                .apkUrl(apkPath)
                .updateTitle("发现新版本V2.2.3")
                .updateContent("1、快来升级最新版本\n2、这次更漂亮了\n3、快点来吧")
                .changeConfig(config)
                .setUpdateDownloadListener(new UpdateDownloadListener() {
                    @Override
                    public void onStart() {
                        Log.e("this", "开始下载");
                    }

                    @Override
                    public void onDownload(int progress) {
                        Log.e("this", "下载中" + progress);
                    }

                    @Override
                    public void onFinish() {
                        Log.e("this", "下载完成");
                    }

                    @Override
                    public void onError(@NotNull Throwable e) {
                        Log.e("this", "下载出错" + e.getMessage());
                    }
                })
                .update(this);
    }

    //通过浏览器下载
    private void update2() {
//        update.UpdateAppUtils.Companion.from(this)
//                .serverVersionCode(2)
//                .serverVersionName("2.0")
//                .apkUrl(apkUrl)
//                .downloadBy(update.UpdateAppUtils.Companion.getDOWNLOAD_BY_BROWSER())
//                .update();
    }

    //强制更新
    private void update3() {

//        update.UpdateAppUtils.
//                .serverVersionCode(2)
//                .serverVersionName("2.0")
//                .apkUrl(apkUrl)
//                .isForce(true)
//                .update();
    }

    //根据versionName判断跟新
    private void update4() {

//        update.UpdateAppUtils
//                .checkBy(update.UpdateAppUtils.Companion.getCHECK_BY_VERSION_NAME())
//                .serverVersionName("2.0")
//                .serverVersionCode(2)
//                .apkUrl(apkUrl)
//                .downloadBy(update.UpdateAppUtils.Companion.getDOWNLOAD_BY_BROWSER())
//                .isForce(true)
//                .update();
    }
}

File: app/src/main/java/com/example/teprinciple/updateappdemo/MainActivity.kt
package com.example.teprinciple.updateappdemo

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.Environment
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView
import constacne.DownLoadBy
import constacne.UiType
import kotlinx.android.synthetic.main.activity_main.*
import listener.OnInitUiListener
import listener.UpdateDownloadListener
import model.UiConfig
import model.UpdateConfig
import update.UpdateAppUtils


class MainActivity : AppCompatActivity() {
    //private val apkUrl = "http://118.24.148.250:8080/yk/update_signed.apk"
    private val apkUrl = "http://minshengyizu.gitee.io/msyz/testappfile/m_1.0.0.apk"
    private val updateTitle = "发现新版本V2.0.0"
    private val updateContent = "1、Kotlin重构版\n2、支持自定义UI\n3、增加md5校验\n4、更多功能等你探索"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 启动应用后删除安装包
        UpdateAppUtils.getInstance().deleteInstalledApk()

        // 基本使用
        btn_basic_use.setOnClickListener {
            UpdateAppUtils
                .getInstance()
                .apkUrl(apkUrl)
                .updateTitle(updateTitle)
                .uiConfig(UiConfig(uiType = UiType.SIMPLE))
                .updateContent(updateContent)
                .update()
        }

        // 浏览器下载
        btn_download_by_browser.setOnClickListener {

            // 使用SpannableString
            val content = SpanUtils(this)
                .appendLine("1、Kotlin重构版")
                .appendLine("2、支持自定义UI").setForegroundColor(Color.RED)
                .appendLine("3、增加md5校验").setForegroundColor(Color.parseColor("#88e16531")).setFontSize(20, true)
                .appendLine("4、更多功能等你探索").setBoldItalic()
                .appendLine().appendImage(R.mipmap.ic_launcher).setBoldItalic()
                .create()

            UpdateAppUtils
                .getInstance()
                .apkUrl(apkUrl)
                .updateTitle(updateTitle)
                .updateContent(content)
                .updateConfig(UpdateConfig().apply {
                    downloadBy = DownLoadBy.APP
                    // alwaysShow = false
                    serverVersionName = "2.0.0"
                })
                .uiConfig(UiConfig(uiType = UiType.PLENTIFUL))
                .update()
        }

        // 自定义UI
        btn_custom_ui.setOnClickListener {
            UpdateAppUtils
                .getInstance()
                .apkUrl(apkUrl)
                .updateTitle(updateTitle)
                .updateContent(updateContent)
                .updateConfig(UpdateConfig(alwaysShowDownLoadDialog = true))
                .uiConfig(UiConfig(uiType = UiType.CUSTOM, customLayoutId = R.layout.view_update_dialog_custom))
                .setOnInitUiListener(object : OnInitUiListener {
                    override fun onInitUpdateUi(view: View?, updateConfig: UpdateConfig, uiConfig: UiConfig) {
                        view?.findViewById<TextView>(R.id.tv_update_title)?.text = "版本更新啦"
                        view?.findViewById<TextView>(R.id.tv_version_name)?.text = "V2.0.0"
                        // do more...
                    }
                })
                .update()
        }

        // java使用示例
        btn_java_sample.setOnClickListener {
            startActivity(Intent(this, JavaDemoActivity::class.java))
        }

        // md5校验
        btn_check_md5.setOnClickListener {
            startActivity(Intent(this, CheckMd5DemoActivity::class.java))
        }

        // 高级使用
        btn_higher_level_use.setOnClickListener {
            // ui配置
            val uiConfig = UiConfig().apply {
                uiType = UiType.PLENTIFUL
                cancelBtnText = "下次再说"
                updateLogoImgRes = R.drawable.ic_update
                updateBtnBgRes = R.drawable.bg_btn
                titleTextColor = Color.BLACK
                titleTextSize = 18f
                contentTextColor = Color.parseColor("#88e16531")
            }

            // 更新配置
            val updateConfig = UpdateConfig().apply {
                force = true
                isDebug = true
                checkWifi = true
                isShowNotification = true
                notifyImgRes = R.drawable.ic_logo
                apkSavePath = Environment.getExternalStorageDirectory().absolutePath + "/teprinciple"
                apkSaveName = "teprinciple"
            }

            UpdateAppUtils
                .getInstance()
                .apkUrl(apkUrl)
                .updateTitle(updateTitle)
                .updateContent(updateContent)
                .updateConfig(updateConfig)
                .uiConfig(uiConfig)
                .setUpdateDownloadListener(object : UpdateDownloadListener {
                    override fun onStart() {
                    }

                    override fun onDownload(progress: Int) {
                    }

                    override fun onFinish() {
                    }

                    override fun onError(e: Throwable) {
                    }
                })
                .update()
        }
    }
}